Mavenizing the project

carlschroedl opened the pull request:
@happyjack27, we had [briefly discussed](https://github.com/happyjack27/autoredistrict/issues/1#issuecomment-171478208) using a formal build system. What do you think about this? 
I've got to call it quits for today, but I can write more about why adopting this tool would be helpful to autoredistrict sometime in the first half of this week if you're curious.

happyjack27 commented:
my main concern is barriers to entry.

i don't want to make it so that someone who doesn't know how to use maven
or install anything or anything like that is unneccessarily burdened.  this
is incredibly simple to build, and i'd like to keep it that way.  i've
found other projects to be a pain in the a$$ to build and i don't want to
become one of those.

this doesn't look like such a barrier, though.  i guess would anyone who
just wants to do a raw build have to do any extra steps or learn anything
new?

On Sun, Jan 31, 2016 at 5:18 PM, Carl Schroedl <notifications@github.com>
wrote:

> @happyjack27 <https://github.com/happyjack27>, we had briefly discussed
> <https://github.com/happyjack27/autoredistrict/issues/1#issuecomment-171478208>
> using a formal build system. What do you think about this?
> I've got to call it quits for today, but I can write more about why
> adopting this tool would be helpful to autoredistrict sometime in the first
> half of this week if you're curious.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/happyjack27/autoredistrict/pull/46
> Commit Summary
>
>    - Adding barebones secure mvn pom.
>    - Moving files to conform to the maven standard directory layout
>    - Ignoring the maven directory for compiled code
>    - Adding MANDIFEST.mf. Adding custom jar as local file dependency.
>    - Updating build and run documentation
>    - Merge branch 'master' into maven
>    - According to travis docs, maven projects are supported out of the
>    box,
>    - Merge branch 'master' of github.com:happyjack27/autoredistrict into
>    maven
>
> File Changes
>
>    - *M* .gitignore
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-0>
>    (1)
>    - *M* .travis.yml
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-1>
>    (2)
>    - *M* README.md
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-2>
>    (15)
>    - *D* build.sh
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-3>
>    (7)
>    - *A* pom.xml
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-4>
>    (92)
>    - *R* src/main/java/com/hexiong/jdbf/DBFReader.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-5>
>    (0)
>    - *R* src/main/java/com/hexiong/jdbf/DBFWriter.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-6>
>    (0)
>    - *R* src/main/java/com/hexiong/jdbf/JDBFException.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-7>
>    (0)
>    - *R* src/main/java/com/hexiong/jdbf/JDBField.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-8>
>    (0)
>    - *R* src/main/java/excel/DialogProgressBar.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-9>
>    (0)
>    - *R* src/main/java/excel/Dispatch.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-10>
>    (0)
>    - *R* src/main/java/excel/ExcelObj.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-11>
>    (0)
>    - *R* src/main/java/excel/ExcelTemplate.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-12>
>    (0)
>    - *R* src/main/java/excel/ExportThread.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-13>
>    (0)
>    - *R* src/main/java/excel/Variant.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-14>
>    (0)
>    - *A* src/main/java/excel/jcom_new.jar
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-15>
>    (0)
>    - *R* src/main/java/geography/DemographicSet.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-16>
>    (0)
>    - *R* src/main/java/geography/Feature.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-17>
>    (0)
>    - *R* src/main/java/geography/FeatureCollection.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-18>
>    (0)
>    - *R* src/main/java/geography/Geometry.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-19>
>    (0)
>    - *R* src/main/java/geography/Project.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-20>
>    (0)
>    - *R* src/main/java/geography/Properties.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-21>
>    (0)
>    - *R* src/main/java/org/autoredistrict/Applet.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-22>
>    (0)
>    - *R* src/main/java/org/autoredistrict/DataAndHeader.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-23>
>    (0)
>    - *R* src/main/java/org/autoredistrict/DialogDownload.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-24>
>    (0)
>    - *R* src/main/java/org/autoredistrict/DialogExport.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-25>
>    (0)
>    - *R* src/main/java/org/autoredistrict/DialogManageLocks.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-26>
>    (0)
>    - *R* src/main/java/org/autoredistrict/DialogMerge.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-27>
>    (0)
>    - *R* src/main/java/org/autoredistrict/DialogMultiColumnSelect.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-28>
>    (0)
>    - *R* src/main/java/org/autoredistrict/DialogSelectLayers.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-29>
>    (0)
>    - *R* src/main/java/org/autoredistrict/DialogShowProperties.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-30>
>    (0)
>    - *R* src/main/java/org/autoredistrict/Download.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-31>
>    (0)
>    - *R* src/main/java/org/autoredistrict/FrameRankedDistricts.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-32>
>    (0)
>    - *R* src/main/java/org/autoredistrict/FrameSeatsVotesChart.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-33>
>    (0)
>    - *R* src/main/java/org/autoredistrict/MainFrame.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-34>
>    (0)
>    - *R* src/main/java/org/autoredistrict/MapPanel.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-35>
>    (0)
>    - *R* src/main/java/org/autoredistrict/PanelGraph.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-36>
>    (0)
>    - *R* src/main/java/org/autoredistrict/PanelGraphDrawArea.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-37>
>    (0)
>    - *R* src/main/java/org/autoredistrict/PanelSeats.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-38>
>    (0)
>    - *R* src/main/java/org/autoredistrict/PanelStats.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-39>
>    (0)
>    - *R* src/main/java/org/autoredistrict/VerticalLabelUI.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-40>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/ShapeFileReader.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-41>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/ValidationPreferences.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-42>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/exception/DataStreamEOFException.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-43>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/exception/InvalidShapeFileException.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-44>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/header/ShapeFileHeader.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-45>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/AbstractShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-46>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/Const.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-47>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/PartType.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-48>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/PointData.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-49>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/ShapeHeader.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-50>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/ShapeType.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-51>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/AbstractMultiPointShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-52>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/AbstractPointShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-53>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/AbstractPolyMShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-54>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/AbstractPolyPlainShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-55>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/AbstractPolyShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-56>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/AbstractPolyZShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-57>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/MultiPatchShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-58>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/MultiPointMShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-59>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/MultiPointPlainShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-60>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/MultiPointZShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-61>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/NullShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-62>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/PointMShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-63>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/PointShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-64>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/PointZShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-65>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/PolygonMShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-66>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/PolygonShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-67>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/PolygonZShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-68>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/PolylineMShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-69>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/PolylineShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-70>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/shape/shapes/PolylineZShape.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-71>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/util/BAUtil.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-72>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/util/DoubleSerializer.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-73>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/util/HexaUtil.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-74>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/util/ISUtil.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-75>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/util/IntSerializer.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-76>
>    (0)
>    - *R*
>    src/main/java/org/nocrala/tools/gis/data/esri/shapefile/util/test.jpage
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-77>
>    (0)
>    - *R* src/main/java/serialization/JSONObject.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-78>
>    (0)
>    - *R* src/main/java/serialization/ReflectionJSONObject.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-79>
>    (0)
>    - *R* src/main/java/serialization/iJSONObject.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-80>
>    (0)
>    - *R* src/main/java/solutions/ClosestPair.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-81>
>    (0)
>    - *R* src/main/java/solutions/District.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-82>
>    (0)
>    - *R* src/main/java/solutions/DistrictMap.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-83>
>    (0)
>    - *R* src/main/java/solutions/Ecology.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-84>
>    (0)
>    - *R* src/main/java/solutions/Edge.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-85>
>    (0)
>    - *R* src/main/java/solutions/Election.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-86>
>    (0)
>    - *R* src/main/java/solutions/Gaussian.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-87>
>    (0)
>    - *R* src/main/java/solutions/Pair.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-88>
>    (0)
>    - *R* src/main/java/solutions/PermIterator.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-89>
>    (0)
>    - *R* src/main/java/solutions/Quadruplet.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-90>
>    (0)
>    - *R* src/main/java/solutions/Settings.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-91>
>    (0)
>    - *R* src/main/java/solutions/StaticFunctions.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-92>
>    (0)
>    - *R* src/main/java/solutions/Test.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-93>
>    (0)
>    - *R* src/main/java/solutions/Triplet.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-94>
>    (0)
>    - *R* src/main/java/solutions/VTD.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-95>
>    (0)
>    - *R* src/main/java/solutions/Vertex.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-96>
>    (0)
>    - *R* src/main/java/solutions/iChangeListener.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-97>
>    (0)
>    - *R* src/main/java/solutions/iDiscreteEventListener.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-98>
>    (0)
>    - *R* src/main/java/solutions/iEvolvable.java
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-99>
>    (0)
>    - *A* src/main/resources/META-INF/MANIFEST.MF
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-100>
>    (1)
>    - *R* src/main/resources/emails/open letter to north carolina
>    endgerrymanderingnow.org.txt
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-101>
>    (0)
>    - *R*
>    src/main/resources/emails/to_california_citizen_redistricting_commision
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-102>
>    (0)
>    - *R*
>    src/main/resources/emails/to_counsel_for_coalition_plaintiffs_for_FL_2015_lawsuit
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-103>
>    (0)
>    - *R* src/main/resources/emails/to_harvard_election_data_archive.txt
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-104>
>    (0)
>    - *R* src/main/resources/emails/to_washington_sos
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-105>
>    (0)
>    - *R*
>    src/main/resources/emails/to_wisconsin_legislative_technology_services.txt
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-106>
>    (0)
>    - *R* src/main/resources/export_template.xls
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-107>
>    (0)
>    - *R* src/main/resources/html_template.txt
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-108>
>    (0)
>    - *R* src/main/resources/icon_set_player.svg
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-109>
>    (0)
>    - *R* src/main/resources/jcom.dll
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-110>
>    (0)
>    - *R* src/main/resources/jcom.jar
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-111>
>    (0)
>    - *R* src/main/resources/ranked_districts.png
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-112>
>    (0)
>    - *R* src/main/resources/representativeness_tooltip.png
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-113>
>    (0)
>    - *R* src/main/resources/seats_votes_asymmetry_tooltip.png
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-114>
>    (0)
>    - *R* src/main/resources/tutorials/graph_heuristic_optimization.png
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-115>
>    (0)
>    - *R* src/main/resources/tutorials/in_practice.docx
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-116>
>    (0)
>    - *R* src/main/resources/tutorials/in_theory.docx
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-117>
>    (0)
>    - *R* src/main/resources/tutorials/powerpoint.docx
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-118>
>    (0)
>    - *R* src/main/resources/tutorials/powerpoint_script
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-119>
>    (0)
>    - *R* src/main/resources/tutorials/samples.docx
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-120>
>    (0)
>    - *R* src/main/resources/tutorials/tutorial_text.txt
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-121>
>    (0)
>    - *R* src/main/resources/voting_power.png
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-122>
>    (0)
>    - *R* src/main/resources/wasted_votes.png
>    <https://github.com/happyjack27/autoredistrict/pull/46/files#diff-123>
>    (0)
>
> Patch Links:
>
>    - https://github.com/happyjack27/autoredistrict/pull/46.patch
>    - https://github.com/happyjack27/autoredistrict/pull/46.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/happyjack27/autoredistrict/pull/46>.
>

happyjack27 commented:
it broke it.  it doesn't build.
you forgot to change the "package ..." at the top of the files.
also now that resources moved that path in the code will have to change too.
and why is there geography now java.geography but ui is now java.org.autoredistrict (and no .ui)?

carlschroedl commented:
Howdy! I'd like to try to reproduce the problem you're describing. Could
you please tell me what commands you were running and what error messages
you observed? On travis and on my workstation, `mvn clean install` is
working fine to build the app.

carlschroedl commented:
To debug this now that it has been reverted, try pulling down the maven
branch from my fork:
http://stackoverflow.com/questions/5884784/how-to-pull-remote-branch-from-somebody-elses-repo

carlschroedl commented:
Also, you might have to re-import the project as a maven project in your
IDE to take advantage of the automated goodness.

carlschroedl commented:
Looks like eclipse supports maven out of the box now:
http://www.eclipse.org/downloads/packages/eclipse-ide-java-ee-developers/mars1
which is cool because a few years back you still had to install a plugin.
Stock Netbeans definitely supports it too.

happyjack27 commented:
no need to debug.  the bug is already known.  the java package heirarchy
wasn't updated.to reflect the refactoring.  for instance, Applet.jar still
says "package ui;" at the top, even though it's now package
main.whatever.whatever.autoredistrict.

i gather that you didn't test build and launch it before commiting because
that would never be able to link right. unless you were just building an
empty directory and running the old jar.


also the ui classes should be in a ui package, and the geography and
solutions packages should be moved inside the autoredistrict package.

and i didn't look close enough to see if it was done or not, but  the java
build and run from command line would need to be changed to reflect the new
locations of the sources (including the run).








On Tue, Feb 2, 2016 at 8:06 AM, Carl Schroedl <notifications@github.com>
wrote:

> Looks like eclipse supports maven out of the box now:
>
> http://www.eclipse.org/downloads/packages/eclipse-ide-java-ee-developers/mars1
> which is cool because a few years back you still had to install a plugin.
> Stock Netbeans definitely supports it too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/happyjack27/autoredistrict/pull/46#issuecomment-178586804>
> .
>

carlschroedl commented:
> no need to debug. the bug is already known. the java package heirarchy wasn't updated.to reflect the refactoring. for instance, Applet.jar still says "package ui;" at the top, even though it's now package main.whatever.whatever.autoredistrict.

Yep, I forgot to update that. Thanks for pointing it out. I apologize for not acknowledging that in my first post-revert response. It turns out that it doesn't matter that the package declarations and import statements haven't been updated, the app still builds and runs using the updated commands that I specified in the README.md, but I will update them nonetheless. The resource names probably still need to be updated. 

> i gather that you didn't test build and launch it before commiting because that would never be able to link right. unless you were just building an empty directory and running the old jar.

I built and launched successfully multiple times using the instructions I placed in the `README.md` before issuing the pull request. Travis also successfully builds.

> also the ui classes should be in a ui package, and the geography and solutions packages should be moved inside the autoredistrict package.

Can do!

> and i didn't look close enough to see if it was done or not, but the java build and run from command line would need to be changed to reflect the new locations of the sources (including the run).

I updated the `README.md` to reflect the new steps to build and run from the command line. I removed the old `build.sh` script, because we don't need it any more with maven. I removed the reference to the `build.sh script` it in the `.travis.yml` file. We don't need to replace it with anything because Travis supports maven natively.

carlschroedl commented:
TLDR; I'll work on the package declarations and imports, find and redefine the resource paths, and update this PR or a subsequent one.

Are you having problems getting the project to build and run when you have the maven branch checked out using the commands in the README? If so, I can help ya out there.

carlschroedl commented on README.md:
```diff
@@ -15,21 +15,16 @@ The -Xmx4096M -Xms1024M arguments tell java to reserve 1GB of memory, and allow
 
```
Here's the goods!

happyjack27 commented:
strange that it builds with the package declarations messed up.  i don't
think that's to spec.  (though i've always though package declarations to
be a bit redundant.)

and even if maven can build it - it should build automatically in a capable
IDE.  personally i have no need for maven or any kind of build scripts like
that 'cause it builds automatically out of the box.  would like to keep it
that way for others, too.  eclipse, at least, can't build without the
package declarations right.

On Tue, Feb 2, 2016 at 9:04 PM, Carl Schroedl <notifications@github.com>
wrote:

> no need to debug. the bug is already known. the java package heirarchy
> wasn't updated.to reflect the refactoring. for instance, Applet.jar still
> says "package ui;" at the top, even though it's now package
> main.whatever.whatever.autoredistrict.
>
> Yep, I forgot to update that. Thanks for pointing it out. I apologize for
> not acknowledging that in my first post-revert response. It turns out that
> it doesn't matter that the package declarations and import statements
> haven't been updated, the app still builds and runs using the updated
> commands that I specified in the README.md, but I will update them
> nonetheless. The resource names probably still need to be updated.
>
> i gather that you didn't test build and launch it before commiting because
> that would never be able to link right. unless you were just building an
> empty directory and running the old jar.
>
> I built and launched successfully multiple times using the instructions I
> placed in the README.md before issuing the pull request. Travis also
> successfully builds.
>
> also the ui classes should be in a ui package, and the geography and
> solutions packages should be moved inside the autoredistrict package.
>
> Can do!
>
> and i didn't look close enough to see if it was done or not, but the java
> build and run from command line would need to be changed to reflect the new
> locations of the sources (including the run).
>
> I updated the README.md to reflect the new steps to build and run from
> the command line. I removed the old build.sh script, because we don't
> need it any more with maven. I removed the reference to the build.sh
> script it in the .travis.yml file. We don't need to replace it with
> anything because Travis supports maven natively.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/happyjack27/autoredistrict/pull/46#issuecomment-178975283>
> .
>

happyjack27 commented:
i haven't tried the commands in the readme.  no need to except if you want
me to verify.  eclipse builds and launches for me.

On Tue, Feb 2, 2016 at 9:11 PM, Carl Schroedl <notifications@github.com>
wrote:

> TLDR; I'll work on the package declarations and imports, find and redefine
> the resource paths, and update this PR or a subsequent one.
>
> Are you having problems getting the project to build and run when you have
> the maven branch checked out using the commands in the README? If so, I can
> help ya out there.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/happyjack27/autoredistrict/pull/46#issuecomment-178978354>
> .
>

happyjack27 commented:
maybe it build because it wasn't cleaned(class files deleted) first?
i don't see how it could possibly link when the import statements are
wrong.  so it stands to reason that it didn't link.

On Tue, Feb 2, 2016 at 9:14 PM, Kevin Baas <happyjack27@gmail.com> wrote:

> i haven't tried the commands in the readme.  no need to except if you want
> me to verify.  eclipse builds and launches for me.
>
> On Tue, Feb 2, 2016 at 9:11 PM, Carl Schroedl <notifications@github.com>
> wrote:
>
>> TLDR; I'll work on the package declarations and imports, find and
>> redefine the resource paths, and update this PR or a subsequent one.
>>
>> Are you having problems getting the project to build and run when you
>> have the maven branch checked out using the commands in the README? If so,
>> I can help ya out there.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/happyjack27/autoredistrict/pull/46#issuecomment-178978354>
>> .
>>
>
>

carlschroedl commented:
Yeah, I'm not sure why it works without updating packages, but it's not for lack of cleaning. As I specified in the readme, I'm running a clean each time. I deleted the bin directory where the former build script put its stuff early in the development on the maven branch, so it wasn't reading stale stuff in from there.

happyjack27 commented:
ok that makes me a little uncomfortable.  it's not breaking when it's
supposed to break.

On Tue, Feb 2, 2016 at 9:38 PM, Carl Schroedl <notifications@github.com>
wrote:

> Yeah, I'm not sure why it works without updating packages, but it's not
> for lack of cleaning. As I specified in the readme, I'm running a clean
> each time. I deleted the bin directory where the former build script put
> its stuff early in the development on the maven branch, so it wasn't
> reading stale stuff in from there.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/happyjack27/autoredistrict/pull/46#issuecomment-178983674>
> .
>

happyjack27 commented:
did you launch it and see that it launched and the gui appeared?  i think
java links at runtime and that's why it didn't break - because it didn't
link.

On Tue, Feb 2, 2016 at 9:40 PM, Kevin Baas <happyjack27@gmail.com> wrote:

> ok that makes me a little uncomfortable.  it's not breaking when it's
> supposed to break.
>
> On Tue, Feb 2, 2016 at 9:38 PM, Carl Schroedl <notifications@github.com>
> wrote:
>
>> Yeah, I'm not sure why it works without updating packages, but it's not
>> for lack of cleaning. As I specified in the readme, I'm running a clean
>> each time. I deleted the bin directory where the former build script put
>> its stuff early in the development on the maven branch, so it wasn't
>> reading stale stuff in from there.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/happyjack27/autoredistrict/pull/46#issuecomment-178983674>
>> .
>>
>
>

carlschroedl commented:
I read ya, I'm surprised it works myself. Yep, I launched the app, and yep, the gui appeared.

carlschroedl commented:
And no exceptions or such like written to stdout.

carlschroedl commented:
Between a few distractions at home and recovering from a corrupted eclipse workspace, I haven't made any progress, but I've got to call it a night. Will take a brief look tomorrow.

happyjack27 commented:
Did you delete the .jar, too?

It occurs to me that: 
1.) deleting the class files does not imply deleting the jar
2.) a build program may terminate with exit code zero simply to report that the build program itself had no bugs, regardless of whether or not the program being built has bugs.  After all, failing to build a program that has compile errors is proper and successful behavior for a build program.

That's all I got.   Probably out of ideas at this point.

I want to add in the new competitiveness stats that FairVote wants but I don't want to create a merge conflict for you.

Sent from my iPhone

> On Feb 2, 2016, at 9:46 PM, Carl Schroedl <notifications@github.com> wrote:
> 
> Between a few distractions at home and recovering from a corrupted eclipse workspace, I haven't made any progress, but I've got to call it a night. Will take a brief look tomorrow.
> 
> —
> Reply to this email directly or view it on GitHub.
>

carlschroedl commented:
Thanks for the suggestions. It looks like I'm going to have to take a break
for a few days to a week, so it might be better to go ahead with those
features and let me deal the merge conflicts later.
